File: elixir/grains/lib/grains.ex
defmodule Grains do
  use Bitwise, only_operators: true
  @doc """
  Calculate two to the power of the input minus one.
  """
  @spec square(pos_integer()) :: {:ok, pos_integer()} | {:error, String.t()}
  def square(number) do
    if number <= 0 or number >= 65 do
      {:error, "The requested square must be between 1 and 64 (inclusive)"}
    else
      {:ok, 1 <<< (number - 1)}
    end
  end

  @doc """
  Adds square of each number from 1 to 64.
  """
  @spec total :: {:ok, pos_integer()}
  def total do
    {:ok, (1 <<< 64) - 1}
  end
end

File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64) throw new ArgumentOutOfRangeException();
        return (ulong)Math.Pow(2.0, (double)(n - 1));
    }

    public static ulong Total()
    {
        return 2*Square(64) - 1;
    }
}